designs/EmbeddedReadmesTechSpec.md

```diff
@@ -0,0 +1,102 @@
+
+First, we need to add a `readme` string element to the [*nuspec* schema](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Packaging/compiler/resources/nuspec.xsd) to ensure that the readme property is recognized and parsed from the *nuspec* file. We also need to ensure that the `PackageReadmeFile` property is parsed from a *csproj* file by editing [NuGet.Build.Tasks.Pack.targets](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L198) to include this line:
+
+```
+Readme="$(PackageReadmeFile)"
+```
+
+Next, we need to add `Readme` member variables to multiple relevant classes and interfaces, and ensure that the Readme property is transmitted from one class to another when `pack` is called. These are:
+* [IPackTaskRequest](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Build.Tasks.Pack/IPackTaskRequest.cs#L16)
+* [PackTask](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs#L14)
+* [PackTaskRequest](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskRequest.cs#L10)
+* [IPackageMetadata](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/IPackageMetadata.cs#L11)
+* [ManifestMetadata](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs#L18)
+* [PackageBuilder](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L25)
+
+In order to transmit the `Readme` value from the source file (*nuspec* or *csproj*) to the final output *nuspec* file, the appropriate assignment statements are needed at the relevant locations.
+
+After this, a `ManifestMetadata` object is used to create the output *nuspec* file that is included in the nupkg. For this step, we need to edit the [ToXElement](https://github.com/NuGet/NuGet.Client/blob/64ed0cb4054226f6060752757d29c50287b312b3/src/NuGet.Core/NuGet.Packaging/PackageCreation/Xml/PackageMetadataXmlExtensions.cs#L32) method in PackageMetadataXmlExtensions.cs to include this line:
+
+```
+AddElementIfNotNull(elem, ns, "readme", metadata.Readme);
+```
+
+**If no Readme property is present in the source file, but a readme.md file exists in the base directory**
+
+Once the properties and files have been processed, if the PackageBuilder object's Readme property is null, we need to look through the base directory for a "readme.md" file. If such a file exists, we must include it in the package and and populate the Readme property with "readme.md".
```

Comment:
I'd caution against this.

There's no precedent to this approach, and there are features with similar behavior, see licenses, icons. 

Users being explicit about metadata is good!

Reply:
There are really 3 pack scenarios. 

1. dotnet pack(msbuild pack). All the content that needs included in the package will be included through Content/None items through MSBuild. We should not do additional searches. We should not pull in things accidentally that might not be part of the project. 
2. nuget.exe pack with a nuspec - without a files element...this basically will include all files under the nuspec directory.
3. nuget.exe pack with a nuspec - with a files element - we will include only the files that the files element has specified. 

In neither of the above 3 cases, do we arbitrarily glob looking for a file to generate metadata for.

Reply:
I've removed this functionality, and the references to it within this spec, based on feedback from the review meeting and Chris' interviews with customers.